### DISCUSSION TITLE
How can I remove all migration history?

### DISCUSSION THREAD

**Discussion by jlei523** [prisma/prisma#16744] [Q&A]
When I run migrate, I'm getting this error:

```
Drift detected: Your database schema is not in sync with your migration history.
```

In order to fix this, I'd have to delete all my data which is not an option. I also do not have a production DB. This is my one and only DB.

How I got here:

1. I removed the primary key and constraints from a table because I needed to bulk insert. The performance was too slow with the keys. I did this directly in the DB and not through migration.
2. When I ran a migration, it gave me the above error.
3. I deleted all my migration files thinking this would reset.
4. Next, I deleted all the entries inside `_prisma_migrations` table.

Somehow, it's still detecting that migrations have differed. Having migration history or migration files is not important to me at this moment. I just want to be able to make schema changes via Prisma.

**Comment by tomhoule** [comment]
> Somehow, it's still detecting that migrations have differed. Having migration history or migration files is not important to me at this moment. I just want to be able to make schema changes via Prisma.

This sentence makes me think about [db push](https://www.prisma.io/docs/guides/database/prototyping-schema-db-push). Have you tried it? Please be aware of the caveats in the docs about the possibility of data loss with certain types of schema changes.

**Comment by janpio** [comment]
When you deleted all migration files, your migration history (now empty) and your database still did not agree. That is what the error message tells you:

> Your database schema is not in sync with your migration history.

You would need to use baselining to fix this: If you are PostgreSQL, documented here for example: https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/relational-databases/baseline-your-database-typescript-postgres Afterwards the SQL in your migration history and your database would be the same, and you could use `migrate dev` again.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
